scripts/script-functions.sh

```diff
@@ -198,6 +266,99 @@ moonfire()
 	esac
 }
 
+sudo_warn()
+{
+	echo_warn -x -p '!!!!!     ' \
+		'------------------------------------------------------------------------------' \
+		'During this script you may be asked to input your root password' \
+		'This is for the purpose of using the sudo command and is necessary to copmlete' \
+		'the script succesfully.' \
+		'------------------------------------------------------------------------------'
+}
+
+
+# Prepare for sqlite directory and set schema into db
+#
+setup_db()
+{
+	DB_NAME=${1:-db}
+	if [ ! -d "${DB_DIR}" ]; then
+		echo_info -x 'Create database directory...'
+		mkdir_moonfire -p "${DB_DIR}"
+	fi
+	DB_PATH="${DB_DIR}/${DB_NAME}"
+	if [ ! -f "${DB_PATH}" ]; then
+		echo_info -x 'Create database and initialize...'
+		sudo -u "${NVR_USER}" -H sqlite3 "${DB_PATH}" < "${SRC_DIR}/schema.sql"
+	fi
+}
+
+# Make sure all sample directories and files owned by moonfire
+#
+fix_ownership()
+{
+	sudo chown -R ${NVR_USER}.${NVR_USER} "$1"
+	echo_info -x "Fix ownership of files in \"$1\"..."
+}
+
+# Make sure samples directory is ready
+#
+prep_sample_file_dir()
+{
+	if [ -z "${SAMPLE_MEDIA_DIR}" ]; then
+		echo_fatal -x "SAMPLE_MEDIA_DIR variable not configured. Check configuration."
+		exit 1
+	fi
+	SAMPLE_FILE_PATH="${SAMPLE_MEDIA_DIR}/${SAMPLE_FILE_DIR}"
+	if [ "${SAMPLE_FILE_PATH##${NVR_HOME}}" != "${SAMPLE_FILE_PATH}" ]; then
+		# Under the home directory, create if not there
+		if [ ! -d "${SAMPLE_FILE_PATH}" ]; then
+			echo_info -x "Created sample file directory: \"$SAMPLE_FILE_PATH\"..."
+			mkdir_moonfire -p "${SAMPLE_FILE_PATH}"
+		fi
+	else
+		if [ ! -d "${SAMPLE_FILE_PATH}" ]; then
+			read_lines <<-MSG1
+	Samples directory $SAMPLE_FILE_PATH does not exist. 
+	If a mounted file system, make sure /etc/fstab is properly configured, 
+	and file system is mounted and directory created.
+	MSG1
+			echo_fatal -L
+		fi
+	fi
+	fix_ownership "${SAMPLE_FILE_PATH}"
+}
+
+# Create user and groups if not there
+#
+prep_moonfire_user()
+{
+	echo_info -x "Create user/group and directories we need..."
+	if ! groupExists "${NVR_GROUP}"; then
+		sudo addgroup --quiet --system ${NVR_GROUP}
+	fi
+	if ! userExists "${NVR_USER}"; then
+		sudo adduser --quiet --system ${NVR_USER} \
+			--ingroup "${NVR_GROUP}" --home "${NVR_HOME}"
+	fi
+	if [ ! -d "${NVR_HOME}" ]; then
+		mkdir_moonfire "${NVR_HOME}"
+	fi
+	sudo chown ${NVR_USER}:${NVR_GROUP} "${NVR_HOME}"
+}
+
+# Correct possible timezone issues
+#
+fix_localtime()
+{
+	if [ ! -L /etc/localtime ] && [ -f /etc/timezone ] &&
+			[ -f "/usr/share/zoneinfo/`cat /etc/timezone`" ]; then
+		echo_info -x "Correcting /etc/localtime setup issue..."
+		sudo rm /etc/localtime
+		sudo ln -s /usr/share/zoneinfo/`cat /etc/timezone` /etc/localtime
```

Comment:
This could replace the rm and ln command:

sudo ln -sf /usr/share/zoneinfo/`cat /etc/timezone` /etc/localtime

the advantage (other than brevity) is that it's atomic, so you can't end up without a /etc/localtime on failure.

I know I should just get issue #12 fixed and obsolete this...